src/compiler/transformers/classFields.ts

```diff
@@ -218,19 +348,47 @@ namespace ts {
         }
 
         function createPrivateIdentifierAccess(info: PrivateIdentifierInfo, receiver: Expression): Expression {
-            receiver = visitNode(receiver, visitor, isExpression);
-            switch (info.placement) {
-                case PrivateIdentifierPlacement.InstanceField:
-                    return context.getEmitHelperFactory().createClassPrivateFieldGetHelper(
-                        nodeIsSynthesized(receiver) ? receiver : factory.cloneNode(receiver),
-                        info.weakMapName
+            return createPrivateIdentifierAccessHelper(info, visitNode(receiver, visitor, isExpression));
+        }
+
+        function createPrivateIdentifierAccessHelper(info: PrivateIdentifierInfo, receiver: Expression): Expression {
+            let getExpression: Expression;
+
+            switch(info.kind) {
+                case PrivateIdentifierKind.Accessor:
+                    getExpression = context.getEmitHelperFactory().createClassPrivateFieldGetHelper(
+                        receiver,
+                        info.brandCheckIdentifier,
+                        info.kind,
+                        info.getterName
                     );
-                default: return Debug.fail("Unexpected private identifier placement");
+                    break;
+                case PrivateIdentifierKind.Method:
+                    getExpression = context.getEmitHelperFactory().createClassPrivateFieldGetHelper(
+                        receiver,
+                        info.brandCheckIdentifier,
+                        info.kind,
+                        info.methodName
+                    );
+                    break;
+                case PrivateIdentifierKind.Field:
+                    getExpression = context.getEmitHelperFactory().createClassPrivateFieldGetHelper(
+                        receiver,
+                        info.brandCheckIdentifier,
+                        info.kind,
+                        info.variableName
+                    );
+                    break;
+                default:
+                    Debug.assertNever(info, "Unknown private element type");
             }
+
+            setCommentRange(receiver, getExpression);
+            return getExpression;
         }
 
         function visitPropertyAccessExpression(node: PropertyAccessExpression) {
-            if (shouldTransformPrivateFields && isPrivateIdentifier(node.name)) {
+            if (shouldTransformPrivateElements && isPrivateIdentifier(node.name)) {
                 const privateIdentifierInfo = accessPrivateIdentifier(node.name);
                 if (privateIdentifierInfo) {
                     return setOriginalNode(

```

Comment:
You can wrap this in a call to `setTextRange(..., node)` to set the correct comment and source map positions using `pos` and `end`.